screens/auth/SignUpScreen.js

```diff
@@ -228,39 +264,135 @@ export default class SignUp extends React.Component {
+    ) {
+      return fieldStateColors.ERROR;
+    } else if (
+      signUpField == signUpFields.PASSWORD &&
+      validate('password', this.state.password)
+    ) {
+      return fieldStateColors.ERROR;
+    } else if (
+      signUpField == signUpFields.NAME &&
+      !this.state.firstName.replace(/\s/g, '').length
+    ) {
+      return fieldStateColors.ERROR;
+    } else {
+      return fieldStateColors.BLURRED;
+    }
+  }
+
+  onFocus(signUpField) {
+    const { indicators } = this.state;
+    indicators[signUpField] = fieldStateColors.FOCUSED;
+    this.setState({
+      indicators
+    });
+  }
+
+  onBlur(signUpField) {
+    const { indicators } = this.state;
+    indicators[signUpField] = this.handleErrorState(signUpField);
+    this.updateErrors();
+    this.setState({
+      indicators
+    });
+  }
+
   render() {
     return (
-      <SignUpContainer>
-        <Input
-          placeholder="First Name"
-          onChangeText={text => this.setState({ firstName: text })}
-          value={this.state.firstName}
-        />
-        <Input
-          placeholder="Last Name"
-          onChangeText={text => this.setState({ lastName: text })}
-          value={this.state.lastName}
-        />
-        <Input
-          placeholder="Phone Number"
-          onChangeText={text => this.setState({ phoneNumber: text })}
-          value={this.state.phoneNumber}
-          keyboardType="number-pad"
-          maxLength={10}
-        />
-        <Input
-          placeholder="Password"
-          secureTextEntry
-          onChangeText={text => this.setState({ password: text })}
-          value={this.state.password}
-        />
-        <Button title="Sign Up" onPress={() => this.handleSubmit()} />
-        <Button
-          title="Already have an account? Log in"
-          onPress={() => this.props.navigation.navigate('Login')}
-        />
-        <Button title="Testing Bypass" onPress={() => this._devBypass()} />
-      </SignUpContainer>
+      <ScrollView showsVerticalScrollIndicator={false}>
+        <AuthScreenContainer>
+          <BigTitle>Sign Up</BigTitle>
+          <FormContainer>
+            <TextFieldContainer>
+              <Caption color={this.state.indicators[signUpFields.NAME]}>
+                Name
+              </Caption>
+              <TextField
+                onBlur={() => this.onBlur(signUpFields.NAME)}
+                onFocus={() => this.onFocus(signUpFields.NAME)}
+                placeholder="Name"
+                onChangeText={text => this.setState({ firstName: text })}
+                value={this.state.firstName}
+                borderColor={this.state.indicators[signUpFields.NAME]}
+              />
+              <InputNoticeContainer>
+                <Caption color={Colors.secondaryText}>
+                  Note: this is how clerks will greet you!
+                </Caption>
+              </InputNoticeContainer>
+            </TextFieldContainer>
```

Comment:
This section of code seems to be repeated a lot (to create the behavior of each of the login/ signin text fields) — it might be worth to create a component that does all of this to make the code easier to read (props to pass in can just be the things that vary per input box, ie the placeholder, the caption, validation functions) -- this will make things that are consistent (ie border color behavior) a lot easier and make the code easier to read